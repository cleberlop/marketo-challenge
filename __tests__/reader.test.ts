import Reader from '../src/helpers/reader';
import fs from 'fs';
jest.mock('fs');
console.log = jest.fn();
const path = __dirname + '/../leads.json';

beforeEach(() => {
  jest.clearAllMocks();
  fs.existsSync = jest.fn().mockReturnValue(true);
});

describe('readInputFile Method', () => {
  const reader = new Reader();

  it('Calls fs.readFileSync', () => {
    expect.assertions(1);
    reader.readInputFile(path);
    expect(fs.readFileSync).toHaveBeenCalledTimes(1);
  });

  it('Reads input file correctly', () => {
    expect.assertions(1);
    fs.readFileSync = jest.fn().mockReturnValue('Correct File');
    const file = reader.readInputFile(path);
    expect(file).toMatchSnapshot();
  });
});
