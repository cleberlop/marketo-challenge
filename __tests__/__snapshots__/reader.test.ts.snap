// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readInputFile Method Reads input file correctly 1`] = `"Correct File"`;
